scripts/pull_github_pr.sh

```diff
@@ -48,6 +48,15 @@ echo -n "Fetching full name of author $PR_LOGIN... "
 USER_NAME=$(curl -s "https://api.github.com/users/$PR_LOGIN" | jq -r .name)
 echo "$USER_NAME"
 
+# Check PR tests status
+PR_HEAD_SHA=$(jq -r .head.sha <<< $PR_DATA)
+PR_TESTS_STATUS=$(curl -s "https://api.github.com/repos/$PROJECT/commits/$PR_HEAD_SHA/status" | jq -r .state)
+if [ "$PR_TESTS_STATUS" != "success" ]; then
+  echo "Some of the tests that ran for this PR were not completed successfully," \
+        "please make sure all tests are done successfully before merge this PR."
+  exit 1
+fi
+
```

Comment:
I'd prefer if this check was done *after* this series is fetched and merged locally (so somewhere after line 74 of this script). This script is used to download the pull request to a local repository, which is perfectly fine, it should just not be pushed to the remote repository later. For instance, somebody might use this script to pull a change locally and check if the errors are reproducible.

Reply:
I was wondering if this script is also used for other purposes beyond merge, thanks for the answer :)
Fixed.

Reply:
@psarna is this fix what you hoped for?
Did you want the test to be done in the very end, go warn the user to not do (manually!) a "git push"? Or like this, in the middle of the code, when the new code is available under some obscure branch and not merged?
I think if we do pull the remote branch but put it into an obscure branch, we should at least print this branch's name together with the error message.

Reply:
I don't remember the details I'm not sure the place where we store the fetch code actually has a name or it's just a temporary head, but I think I'm suggesting that if we leave it in any other place but HEAD, we should say where we left it, otherwise the user won't know how to use it anyway, and we can just move the check to the beginning of the script.

Another question is whether the runner of this script really wants to use the fetched version as-is, or use it after the merge into his current "next". Since the CI code does the latter, I think the person who uses the script would also like the latter - and this is not what the script currently does... It exists before doing the merge, IIUC.

Reply:
> I don't remember the details I'm not sure the place where we store the fetch code actually has a name or it's just a temporary head, but I think I'm suggesting that if we leave it in any other place but HEAD, we should say where we left it, otherwise the user won't know how to use it anyway, and we can just move the check to the beginning of the script.

The script fetches the PR but without creating a local branch. the PR changes can still be accessible by `git log HEAD..FETCH_HEAD` - I'll print it to the console